### DISCUSSION TITLE
[1.9] 2023-10-15

### DISCUSSION THREAD

**Discussion by darthcloud** [darthcloud/BlueRetro#804] [General]
# Changelog

## New features
* VMU emulation for Dreamcast. (thanks @breademan)
  See Docs:
  * [Web config -> DC VMU Manager page](https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-BLE-Web-Config-User-Manual#72---dc-vmu-manager-page)
  * [Web config -> Output config option for DC](https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-System-Specific-User-Manual#accessories-15)
  * [Web config -> Output config](https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-BLE-Web-Config-User-Manual#23---output-config)
* Macro functions buttons are now customizable.
  See Docs:
  * [Button combinations functions](https://github.com/darthcloud/BlueRetro/wiki#6---button-combinations-functions)
* PlayStation Analog mode status LED for P1 & P2 on port LED 3 & 4.
  See Docs:
  * [Port Status LED](https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-HW2-Internal-Install-Specification#7---port-status-led)
* Toggle between GamePad & GamePadAlt config without reboot or config change via Macro (All system) or Home buttons (PCE, Genesis, Saturn, Dreamcast & 3DO only).
  See Docs:
  * [Button combinations functions](https://github.com/darthcloud/BlueRetro/wiki#6---button-combinations-functions)
* Support for 8BitDo N64 Modkit D mode.
  See Docs:
  * [8BitDo N64 Modkit paring guide](https://github.com/darthcloud/BlueRetro/wiki/Controller-pairing-guide#15---8bitdo-n64-modkit-paring-guide)

# Bug fixes
* Various BT connection issues introduced in v1.8.4.
* BT keyboard random input.
* Parallel system config override not applying.
* Activision games PS2 support (CoD 2 & 3 now work).
* All port controlled by a single controller gets enable on hot plug systems.
* Atari Jaguar FW not working
* Fix PC-FX timing (random input in some games)
* Fix autofire/turbo on PS1/PS2

# Firmware is available in 2 different flavor
## 1. Regular FW as before retroactively renamed HW1 (Typically for external adapter)
Use firmwares from HW1 archive for:
* [WillConsole's](https://twitter.com/WillConsole) PSUnoRetro
* All [Humble Bazooka's](https://twitter.com/humblebazooka) BT dongle
* [RetroTime's](https://twitter.com/bixxewoscht) N64 dongle
* [RetroOnyx's](https://twitter.com/mellott124) VB dongle
* [GrechTech's](https://twitter.com/GrechTech) RetroRosetta
* any DIY adapter you made base on the original BlueRetro spec
* most likely anything from aliexpress

## 2. HW2 FW that support power management and port detection (Typically for internal install)
Use firmwares from HW2 archive for:
* [Laser Bear Industries's](https://twitter.com/collingall) GameCube BlueRetro Internal Adapter
* any DIY internal install you made base on the HW2 spec

## [OTA Firmware update page](https://blueretro.io/ota.html) has protection to prevent cross flashing between HW1 & HW2 firmware.

# Download latest firmware archives from https://darthcloud.itch.io/blueretro

<hr /><em>This discussion was created from the release <a href='https://github.com/darthcloud/BlueRetro/releases/tag/v1.9'>[1.9] 2023-10-15</a>.</em>

**Comment by YgorSan** [comment]
Guitar Hero 3 worked before but I did a quick test here and it seems to be working well in COD 3, thanks for your contribution to the community, I've been eyeing it for a while and I'm happy with this and the other new features

**Reply by darthcloud** [reply]
Are you saying guitar hero 3 used to work in previous version and now it doesn't work in v1.9??

**Comment by golimonkey** [comment]
Hi, first of all ty for your great work on this project. Is it possible to implement analog triggers for DS4?

**Comment by EvertonDuarte001** [comment]
Atualizei pra essa versão e os cod 2 e 3 funcionaram perfeitamente, o guita hero 3 tbm, o amigo no outro comentário disse que parecia que não estava funcionando, no meu funcionou tbm, agora posso aproveitar esses dois jogos com tranquilidade, obrigado
